plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -105,24 +109,68 @@
 
     // wait for any post-import operations too
     waitUntilIdle();
+    assertFalse("imported projects have errors", projectsHaveErrors(projects));
 
     return projects;
   }
 
-  /** Wait for any spawned jobs to complete (e.g., validation jobs). */
-  public static void waitUntilIdle() {
-    while (!Job.getJobManager().isIdle()) {
-      Display display = Display.getCurrent();
-      if (display != null) {
-        while (display.readAndDispatch()) {
-          /* spin */
+  /**
+   * Return true if projects have errors; lists any found to <code>System.err</code>.
+   */
+  public static boolean projectsHaveErrors(Collection<IProject> projects) {
+    List<String> foundProblems = new ArrayList<>();
+    for (IProject project : projects) {
+      try {
+        IMarker[] problems = project.findMarkers(IMarker.PROBLEM, true /* includeSubtypes */,
+            IResource.DEPTH_INFINITE);
+        for (IMarker problem : problems) {
+          int severity = problem.getAttribute(IMarker.SEVERITY, IMarker.SEVERITY_INFO);
+          if (severity >= IMarker.SEVERITY_ERROR) {
+            foundProblems.add(formatProblem(problem));
+          }
         }
+      } catch (CoreException ex) {
+        // re-throw as we shouldn't have problems in tests
+        throw new RuntimeException(ex);
       }
-      try {
+    }
+    if (foundProblems.isEmpty()) {
+      return false;
+    }
+    System.err.println("Project errors:");
+    for (String problem : foundProblems) {
+      System.err.println(" " + problem);
+    }
+    return true;
+  }
+
+  private static String formatProblem(IMarker problem) {
+    StringBuilder sb = new StringBuilder();
+    sb.append(problem.getResource().getFullPath());
+    sb.append(':');
+    sb.append(problem.getAttribute(IMarker.LINE_NUMBER, -1));
+    sb.append(": ");
+    sb.append(problem.getAttribute(IMarker.MESSAGE, ""));
+    return sb.toString();
+  }
+
+  /** Wait for any spawned jobs and builds to complete (e.g., validation jobs). */
+  public static void waitUntilIdle() {
+    try {
+      Job.getJobManager().join(ResourcesPlugin.FAMILY_MANUAL_BUILD, null);
+      Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
+
+      while (!Job.getJobManager().isIdle()) {
+        Display display = Display.getCurrent();
+        if (display != null) {
+          while (display.readAndDispatch()) {
+            /* spin */
+          }
+        }
         Thread.sleep(10);
```

Comment:
Ideally it shouldn't be necessary at all, since we first wait for any build jobs.  But I should rework this to `join()` on the build families within the loop.